Library/Homebrew/os/linux/elf.rb

```diff
@@ -127,7 +130,20 @@ def initialize(path)
     private
 
     def needed_libraries(path)
-      if DevelopmentTools.locate "readelf"
+      if HOMEBREW_PATCHELF_RB
+        patcher = path.patcher
+        soname = begin
+            patcher.soname
+          rescue PatchELF::PatchError
```

Comment:
It makes me a bit nervous to mask all `PatchELF` errors like this. Can we print a warning message?